lib/rspec/support/object_inspector.rb

```diff
@@ -3,23 +3,51 @@ module Support
     # Provide additional output details beyond what `inspect` provides when
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
+      DEFAULT_OPTIONS = {
+        :nested => false,
+        :joiner => ',',
+        :braces => true,
+        :trailing_comma => false
+      }
       # @api private
-      def self.inspect(object)
-        if Time === object
+      # rubocop:disable CyclomaticComplexity
+      # rubocop:disable MethodLength
+      def self.inspect(object, options=DEFAULT_OPTIONS)
+        case object
+        when Time
           format_time(object)
-        elsif defined?(DateTime) && DateTime === object
-          format_date_time(object)
-        elsif defined?(BigDecimal) && BigDecimal === object
-          "#{object.to_s 'F'} (#{object.inspect})"
-        elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
-          object.description
+        when Hash
+          format_hash(object, options)
```

Comment:
@myronmarston

The way arrays are handled is based on how `pp` works, and it gets quite finicky especially when the length of characters you want to display is over a certain number.

I believe that if we are going to move it into the `ObjectInspector`, we really shouldn't be trying to replicate the behaviour of `pp` inside it. However, doing that we will require some discussion about how we want to display the output.

I don't think that I should go down the path of writing a solution, without having a discussion about it first, as it will be based purely on my opinion of how the output should be handled.